lib/ddtrace/pipeline/span_filter.rb

```diff
@@ -18,12 +18,12 @@ def call(trace)
         deleted = Set.new
 
         trace.delete_if do |span|
-          if deleted.include?(span.parent)
-            deleted << span
+          if deleted.include?(span.parent_id)
+            deleted << span.span_id
             true
           else
             drop = drop_it?(span)
-            deleted << span if drop
+            deleted << span.span_id if drop
             drop
           end
```

Comment:
Tiny question: Would it make sense to make `deleted` a map (or a set) to enable faster `deleted.include?` lookups?

Also minor: I guess this could be simplified as

```ruby
trace.delete_if do |span|
  should_delete = deleted.include?(span.parent_id) || drop_it?(span)
  deleted << span.span_id if should_delete
  should_delete
end
```

Reply:
👍